### PR TITLE
doc: matter: add dev environment section

### PR DESCRIPTION
**Pull request by greg-fer**
Expand Configuration in NCS section with OTBR info.

Signed-off-by: Grzegorz Ferenc <Grzegorz.Ferenc@nordicsemi.no>

### CODE REVIEW

File: doc/nrf/ug_matter.rst [hunk 1]
```diff
@@ -50,32 +50,122 @@ The |NCS|'s Multiprotocol Service Layer (MPSL) driver allows running Bluetooth L
    nRF Connect platform in Matter
 
 For detailed description, see the `nRF Connect platform overview`_ Matter documentation page.
+If you are new to Matter, check also the tutorials on `DevZone`_.
+
+Matter development environment
+******************************
+
+When you program any of the available Matter samples to the supported development kits, you automatically build Matter's nRF Connect platform.
```
Comment (Damian-Nordic) [inline doc/nrf/ug_matter.rst:58]: I'm not sure what "Matter's nRF Connect platform" refers to. If it means that the platform integration component in Matter then it's not enough to say that DK + Matter's nRF Connect platform form the Matter accessory. How about using "Matter  stack for nRF Connect platform"? E.g. "Building any of the available Matter samples involves building Matter stack for nRF Connect platform".

File: doc/nrf/ug_matter.rst [hunk 2]
```diff
@@ -50,32 +50,122 @@ The |NCS|'s Multiprotocol Service Layer (MPSL) driver allows running Bluetooth L
...
+The following figure shows the protocol types and the available Matter controllers.
+
+.. figure:: images/matter_protocols_controllers.svg
+   :scale: 50%
+   :alt: Protocol types and controllers used by Matter
+
+   Protocol types and controllers used by Matter
+
+To enable the IPv6 communication with the Matter accessory device over the Thread network, the Matter controller requires the Thread Border Router.
+This is because the Matter controller does not have an 802.15.4 Thread interface.
```
Comment (Damian-Nordic) [inline doc/nrf/ug_matter.rst:79]: pendantic: In theory the Matter controller can be a Thread device as well, so a more accurate statement would be that the Matter controller **typically** doesn't have an 802.15.4 Thread interface.

**Review by Damian-Nordic — COMMENTED** [review]
Very nice description. Just a few suggestions from me.

File: doc/nrf/ug_matter.rst (hunk 1 above)
Comment (LuDuda) [reply doc/nrf/ug_matter.rst:58]: Personally, I like your last proposal.

File: doc/nrf/ug_matter.rst [hunk 3]
```diff
@@ -50,32 +50,122 @@ The |NCS|'s Multiprotocol Service Layer (MPSL) driver allows running Bluetooth L
...
 For detailed description, see the `nRF Connect platform overview`_ Matter documentation page.
+If you are new to Matter, check also the tutorials on `DevZone`_.
+
+Matter development environment
+******************************
+
+When you program any of the available Matter samples to the supported development kits, you automatically build Matter's nRF Connect platform.
+The development kit and Matter's nRF Connect platform that is programmed on the development kit form the Matter accessory device.
+
+The default use case of a Matter accessory device is within a Matter network that is built on top of a low-power, 802.15.4-compatible Thread network.
```
[the remark this answers is not in the document]
Comment (LuDuda) [reply doc/nrf/ug_matter.rst:61]: I think it makes sense to add "For nRF Connect SDK, the default..".

File: doc/nrf/ug_matter.rst (hunk 2 above)
Comment (LuDuda) [reply doc/nrf/ug_matter.rst:79]: I agree, in original docs, there was a word "usually". Let's be future-proof then ;)

File: doc/nrf/ug_matter.rst [hunk 4]
```diff
@@ -50,32 +50,122 @@ The |NCS|'s Multiprotocol Service Layer (MPSL) driver allows running Bluetooth L
...
+A Matter controller is a role within the Matter environment.
+This controller interacts with the accessory devices using the following protocols:
+
+* Bluetooth LE during the commissioning process - to securely pass the network credentials and provision the accessory device into the Thread network.
+* Regular IPv6 communication after the accessory device joins the Thread network - to interact with each other by exchanging application messages.
+  For example, to report temperature measurements of a sensor.
+
+The following figure shows the protocol types and the available Matter controllers.
+
+.. figure:: images/matter_protocols_controllers.svg
```
Comment (LuDuda) [inline doc/nrf/ug_matter.rst:72]: This image needs label replacement. Just commenting to remember to replace it.

File: doc/nrf/ug_matter.rst [hunk 5]
```diff
@@ -50,32 +50,122 @@ The |NCS|'s Multiprotocol Service Layer (MPSL) driver allows running Bluetooth L
...
 
-Configuration in |NCS|
-**********************
+Matter development environment setup options
+============================================
 
-Matter's nRF Connect platform is built when programming any of the available :ref:`matter_samples` to the supported development kits, which creates a Matter accessory device.
-Such a device requires additional configuration if it is to be paired and controlled remotely over a Project CHIP network built on top of a low-power, 802.15.4 Thread network.
-The following figure shows an overview of the network topology with the required configuration elements.
+You can run the Matter controller on the same device as the Thread Border Router or run the Matter controller and the Thread Border Router on different devices.
```
Comment (kkasperczyk-no) [inline doc/nrf/ug_matter.rst:96]: nit: How about different -> separate?

**Review by kkasperczyk-no — COMMENTED** [review]
Good job, it looks really nice overall.

File: doc/nrf/ug_matter.rst (hunk 2 above)
Comment (greg-fer) [reply doc/nrf/ug_matter.rst:79]: Can we be more specific what is hiding behind "typically"/"usually"?

File: doc/nrf/ug_matter.rst (hunk 1 above)
Comment (greg-fer) [reply doc/nrf/ug_matter.rst:58]: Implemented.

File: doc/nrf/ug_matter.rst (hunk 4 above)
Comment (greg-fer) [reply doc/nrf/ug_matter.rst:72]: Implemented the new one.

**Review by kkasperczyk-no — APPROVED** [review]

**Review by Damian-Nordic — APPROVED** [review]

**Review by LuDuda — APPROVED** [review]

**Comment by rlubos** [comment]
We should add `--exclude doc/nrf/images` to `checkpatch.conf`, to avoid *.svg files being processed.

File: doc/nrf/links.txt [hunk 6]
```diff
@@ -87,13 +87,12 @@
...
-.. _`Building and installing Android CHIPTool`: https://github.com/nrfconnect/sdk-connectedhomeip/blob/b35e8e/docs/guides/nrfconnect_android_commissioning.md#user-content-building-and-installing-android-chiptool
-.. _`Configuring PC as Thread Border Router`: https://github.com/nrfconnect/sdk-connectedhomeip/blob/b35e8e/docs/guides/nrfconnect_android_commissioning.md#user-content-configuring-pc-as-thread-border-router
-.. _`Preparing and commissioning accessory device`: https://github.com/nrfconnect/sdk-connectedhomeip/blob/b35e8e/docs/guides/nrfconnect_android_commissioning.md#user-content-preparing-accessory-device
+.. _`Working with Python Controller`: https://github.com/project-chip/connectedhomeip/blob/master/src/controller/python/README.md
 .. _`Matter Protocol Overview`: https://github.com/nrfconnect/sdk-connectedhomeip/blob/b35e8e/README.md
 .. _`nRF Connect platform overview`: https://github.com/nrfconnect/sdk-connectedhomeip/blob/b35e8e/docs/guides/nrfconnect_platform_overview.md
+.. _`other controller setups`: https://github.com/project-chip/connectedhomeip/tree/master/src/controller
```
Comment (LuDuda) [inline doc/nrf/links.txt:95]: @greg-fer This link should point to sdk-connectedhomeip repo instead. We will update the repo affer new documentation is in.

**Review by rlubos — APPROVED** [review]

**Pull request merged**